# Preprocessing

In [4]:
import pandas as pd

In [5]:
pisa = pd.read_csv('pisa2012.csv',encoding = "ISO-8859-1")
pisa.head(2)

c:\users\amous\appdata\local\programs\python\python39\lib\site-packages\IPython\core\interactiveshell.py:3146: DtypeWarning: Columns (15,16,17,21,22,23,24,25,26,30,31,36,37,45,65,123,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,284,285,286,287,288,289,290,291,292,293,294,295,296,297,298,299,300,301,302,303,307,308,309,310,311,312,313,314,315,316,317,318,319,320,321,322,323,324,325,326,327,328,329,330,331,332,333,334,335,336,337,338,339,340,341,342,343,344,345,346,347,348,349,350,351,352,353,354,355,356,357,376,377,378,379,380,381,382,383,384,385,386,387,388,389,390,391,392,393,394,395,396,397,398,399,400,401,402,403,475) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


Unnamed: 0      CNT  SUBNATIO  STRATUM      OECD       NC  SCHOOLID  \
0           1  Albania     80000  ALB0006  Non-OECD  Albania         1   
1           2  Albania     80000  ALB0006  Non-OECD  Albania         1   

   STIDSTD  ST01Q01  ST02Q01  ...  W_FSTR75  W_FSTR76 W_FSTR77 W_FSTR78  \
0        1       10      1.0  ...   13.7954   13.9235  13.1249  13.1249   
1        2       10      1.0  ...   13.7954   13.9235  13.1249  13.1249   

   W_FSTR79 W_FSTR80 WVARSTRR VAR_UNIT SENWGT_STU  VER_STU  
0    4.3389  13.0829       19        1     0.2098  22NOV13  
1    4.3389  13.0829       19        1     0.2098  22NOV13  

[2 rows x 636 columns]

In [6]:
pisa.shape

(485490, 636)

### What is the structure of your dataset?

* it contains 636 columns 

### What is/are the main feature(s) of interest in your dataset?

* After a look at the features, I'm interested in the student behaviors columns and what might affect those behaviors

### What features in the dataset do you think will help support your investigation into your feature(s) of interest?

* I will select family features, teacher support, and some student attitudes

In [11]:
selected_columns = ['AGE','NC','ST04Q01',
           'ST11Q01','ST11Q02','ST11Q03','ST11Q04',
           'ST19Q01','ST15Q01',
           'ST44Q03','ST44Q04','ST44Q07','ST44Q08',
           'ST87Q01','ST87Q04','ST87Q06','ST87Q09',
           'TEACHSUP','MATWKETH']

df = pisa[selected_columns]
df.head()

AGE       NC ST04Q01 ST11Q01 ST11Q02 ST11Q03 ST11Q04  \
0  16.17  Albania  Female     Yes     Yes     Yes     Yes   
1  16.17  Albania  Female     Yes     Yes     NaN     Yes   
2  15.58  Albania  Female     Yes     Yes      No     Yes   
3  15.67  Albania  Female     Yes     Yes      No     Yes   
4  15.50  Albania  Female     Yes     Yes     Yes     NaN   

                        ST19Q01                             ST15Q01  \
0   Working part-time <for pay>  Other (e.g. home duties, retired)    
1  Working full-time <for pay>         Working full-time <for pay>    
2  Working full-time <for pay>         Working full-time <for pay>    
3  Working full-time <for pay>         Working full-time <for pay>    
4   Working part-time <for pay>         Working part-time <for pay>   

           ST44Q03          ST44Q04          ST44Q07          ST44Q08  \
0  Slightly likely           Likely           Likely    Very   Likely   
1  Slightly likely  Slightly likely  Slightly likely           Likely   
2           Likely    Very   Likely    Very   Likely  Slightly likely   
3              NaN              NaN              NaN              NaN   
4           Likely           Likely  Slightly likely    Very   Likely   

             ST87Q01            ST87Q04            ST87Q06         ST87Q09  \
0              Agree     Strongly agree  Strongly disagree           Agree   
1                NaN                NaN                NaN             NaN   
2                NaN                NaN                NaN             NaN   
3                NaN                NaN                NaN             NaN   
4  Strongly disagree  Strongly disagree  Strongly disagree  Strongly agree   

   TEACHSUP  MATWKETH  
0      1.68    0.2882  
1       NaN    0.6490  
2       NaN    2.0389  
3      1.68       NaN  
4      0.11    1.6881

In [12]:
new_names = ['age', 'country', 'gender',
            'mother', 'father', 'brothers','sisters',
            'father_wk_st','mother_wk_st',
            'teacher_not_explaining_well','bad_guesses','teacher_didnt_get_students_interested','unlucky',
            'outsider','awkward','lonely','satisfied_at_school',
            'teacher_support','math_wk_ethic']
column_names = {selected_columns[i]: new_names[i] for i in range(len(selected_columns))} 

df = df.rename(columns=column_names)

In [13]:
df.to_csv('selected_columns_from_pisa.csv', index=False)